client/core/trade.go

```diff
@@ -1526,6 +1628,7 @@ func (c *Core) tick(t *trackedTrade) (assetMap, error) {
 			swaps = append(swaps, match)
 			sent += match.Quantity
 			quoteSent += calc.BaseToQuote(match.Rate, match.Quantity)
+			t.updateFeesPaid(c.ctx, match, true)
```

Comment:
Redeems are batched, so multiple matches may use the same transaction. This needs to be checked per CoinID instead of per match.

Reply:
I probably don't understand correctly, but it does check per coinID I think. It check's on the `match.MetaData.Proof.TakerSwap` or `match.MetaData.Proof.TakerRedeem` coin ID

Do you mean that the tx fee burden may be spaced over several matches?

Hmm, yeah. Should we make an effort to divide it up based on weight if one tx would init or redeem several matches? Currently it might end up in a larger than actual value if you were doing strict accounting across all of your swaps.

Reply:
> Redeems are batched, so multiple matches may use the same transaction. This needs to be checked per CoinID instead of per match.

I want to avoid `*Core` not doing any coin ID inspection. Just running each match separately.

Reply:
If two matches are redeemed by the same coin ID, then the current code will count double the fees. If we're avoiding any CoinID inspection in Core then the I guess the only way is to check how many redemptions were done by the tx in the wallet, and then divide the transaction fee by that amount.

Reply:
The double counting applies to all assets, not just dynamic tx assets, so it's a bug that would need solving for say DCR and BTC too.

Reply:
Oh nevermind it's not an issue in the other assets.

Reply:
Chatting on Matrix a bit, a suggestion might be to update `client/db.MatchProof` with a `SwapBatchSize uint32` that we set in `swapMatchGroup` where we know the batch size (and similar for redeem).

I think it lets us continue to iterate through matches like now and just `+= actualFees / batchSize`.   Iterating over `batchSize` matches, we get the desired effect of just adding `actualFees` instead of `actualFees * batchSize`.

Reply:
Oh, I see now. The order may have the same swap counting multiple times even though it was only one transaction.

Reply:
Right.  Long convo in matrix ongoing, but I think we're gravitating toward:

```go
TransactionFeesPaid(ctx context.Context, coinID dex.Bytes) (fees uint64, secretHashes [][]byte, err error)
```

And then the caller knows which match index in the batch they are, plus the number of swaps in the batch.

Then the fees can be added different ways:
- add the full amount for just the leader swap (matching first secret hash) and zero for the rest
- OR do division by the batch size and have the leader also add any remainder

Reply:
If it's the same coinid can we just use that to ensure that tx has only been added once for the order?

I forget if matches from different orders can be grouped together and have the same init/redeem tx

Reply:
> If it's the same coinid can we just use that to ensure that tx has only been added once for the order?

@buck54321 doesn't want core interpreting coin IDs, and I agree 

> I forget if matches from different orders can be grouped together and have the same init/redeem tx

They cannot.

Reply:
Ok. So will need to check on the tx each time? no cache?

Reply:
We can pretend geth is cacheing everything.

Reply:
RIP geth